geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/MinMax.scala

```diff
@@ -23,114 +24,162 @@ import scala.reflect.ClassTag
     def min: T
     def max: T
-    def minmax(value: T, min: T, max: T): (T, T)
+    def min(left: T, right: T): T
+    def max(left: T, right: T): T
   }
 
-  abstract class ComparableMinMax[T <: Comparable[T]] extends MinMaxDefaults[T] {
-    override def minmax(value: T, min: T, max: T): (T, T) = {
-      val mn = if (value.compareTo(min) > 0) min else value
-      val mx = if (value.compareTo(max) < 0) max else value
-      (mn, mx)
+  object MinMaxDefaults {
+    def apply[T](binding: Class[_]): MinMaxDefaults[T] = {
+      if (binding == classOf[String]) {
+        MinMaxString.asInstanceOf[MinMaxDefaults[T]]
+      } else if (binding == classOf[Integer]) {
+        MinMaxInt.asInstanceOf[MinMaxDefaults[T]]
+      } else if (binding == classOf[java.lang.Long]) {
+        MinMaxLong.asInstanceOf[MinMaxDefaults[T]]
+      } else if (binding == classOf[java.lang.Float]) {
+        MinMaxFloat.asInstanceOf[MinMaxDefaults[T]]
+      } else if (binding == classOf[java.lang.Double]) {
+        MinMaxDouble.asInstanceOf[MinMaxDefaults[T]]
+      } else if (classOf[Date].isAssignableFrom(binding)) {
+        MinMaxDate.asInstanceOf[MinMaxDefaults[T]]
+      } else if (classOf[Geometry].isAssignableFrom(binding)) {
+        MinMaxGeometry.asInstanceOf[MinMaxDefaults[T]]
+      } else {
+        throw new IllegalArgumentException(s"No implicit default available for type: $binding")
+      }
     }
   }
 
+  abstract class ComparableMinMax[T <: Comparable[T]] extends MinMaxDefaults[T] {
+    override def min(left: T, right: T): T = if (left.compareTo(right) > 0) right else left
+    override def max(left: T, right: T): T = if (left.compareTo(right) < 0) right else left
+  }
+
   implicit object MinMaxString extends ComparableMinMax[String] {
-    override val min: String = "~~~"
-    override val max: String = ""
+    override val min: String = ""
+    override val max: String = "~~~"
   }
 
-  implicit object MinMaxInt extends ComparableMinMax[java.lang.Integer] {
-    override val min: java.lang.Integer = java.lang.Integer.MAX_VALUE
-    override val max: java.lang.Integer = java.lang.Integer.MIN_VALUE
+  implicit object MinMaxInt extends ComparableMinMax[Integer] {
+    override val min: Integer = Integer.MIN_VALUE
+    override val max: Integer = Integer.MAX_VALUE
   }
 
   implicit object MinMaxLong extends ComparableMinMax[java.lang.Long] {
-    override val min: java.lang.Long = java.lang.Long.MAX_VALUE
-    override val max: java.lang.Long = java.lang.Long.MIN_VALUE
+    override val min: java.lang.Long = java.lang.Long.MIN_VALUE
+    override val max: java.lang.Long = java.lang.Long.MAX_VALUE
   }
 
   implicit object MinMaxFloat extends ComparableMinMax[java.lang.Float] {
-    override val min: java.lang.Float = java.lang.Float.MAX_VALUE
-    override val max: java.lang.Float = java.lang.Float.MIN_VALUE
+    override val min: java.lang.Float = java.lang.Float.MIN_VALUE
+    override val max: java.lang.Float = java.lang.Float.MAX_VALUE
```

Comment:
It was just confusing... they were being set to these values as initial markers. i switched the logic to set the min to the max and check for that value in the MinMax itself. E.g. lines 34, 103, 106 etc